tests/courses.test.js

```diff
@@ -0,0 +1,121 @@
+const { assert } = require('chai');
+const requests = require('./utils/requests');
+const { knex, cleanDb } = require('./utils/db');
+
+// Starts the app
+require('../src/app.js');
+
+describe('Integration courses tests', () => {
+  let response;
+  const fakeToken = 'diego';
+
+  before(() => cleanDb());
+  afterEach(() => cleanDb());
+
+  describe('Get courses', () => {
+    describe('When there are courses', () => {
+      let expectedCourses;
+      // Set up database
+      beforeEach(async () => {
+        expectedCourses = [
+          {
+            id: 'coursename2',
+            name: 'course name 2',
+            description: 'course description 2',
+          },
+          {
+            id: 'coursename1',
+            name: 'course name 1',
+            description: 'course description 1',
+          },
+        ];
+
+        await knex('courses').del();
+        await knex('courses').insert([
+          {
+            id: 'coursename1',
+            name: 'course name 1',
+            description: 'course description 1',
+          },
+          {
+            id: 'coursename2',
+            name: 'course name 2',
+            description: 'course description 2',
+          },
+          {
+            id: 'coursename3',
+            name: 'course name 3',
+            description: 'course description 3',
+          },
+        ]);
+        await knex('course_users').del();
+        await knex('course_users').insert([
+          {
+            course_id: 'coursename2',
+            user_id: 'diego',
+            role: 'admin'
+          },
+          {
+            course_id: 'coursename1',
+            user_id: 'diego',
+            role: 'admin'
+          },
+          {
+            course_id: 'coursename3',
+            user_id: 'pedro',
+            role: 'admin'
+          },
+          {
+            course_id: 'coursename3',
+            user_id: 'joaco',
+            role: 'student'
+          },
+        ]);
+        await knex('guides').del();
+        await knex('guides').insert([
+          {
+            course_id: 'coursename',
+            guide_id: 'guide id',
+            description: 'description',
+          },
+        ]);
+      });
+
+      beforeEach(async () => {
+        response = await requests.getCourses({ token: fakeToken });
+      });
+
+      it('status is OK', () => assert.equal(response.status, 200));
+
+      it('body has the course', () => assert.deepEqual(response.body, expectedCourses));
+    });
+  });
+
+  describe('Add course', () => {
+    let courseToBeAdded;
+
+    beforeEach(() => {
+      const name = 'curso';
+      const description = 'un curso mas';
+      courseToBeAdded = { name, description, id: name };
+    });
+
+    describe('When is successfully added', () => {
```

Comment:
add tests for border cases (I really would add some of them because that will help us is some weeks when we change somethings in the projet)

you could check: https://github.com/math-learning/users-service/blob/master/tests/users.test.js

Reply:
I totally agree, after finishing the happy path I will do some border case tests